frontend/packages/console-dynamic-plugin-sdk/src/utils/fetch/console-fetch.ts

```diff
@@ -27,22 +27,29 @@ export const consoleFetch: ConsoleFetch = async (url, options = {}, timeout = 60
   return Promise.race([fetchPromise, timeoutPromise]);
 };
 
+const parseData = async (response) => {
+  const text = await response.text();
+  const isPlainText = response.headers.get('Content-Type') === 'text/plain';
+  if (!text) {
+    return isPlainText ? '' : {};
+  }
+  return isPlainText || !response.ok ? text : JSON.parse(text);
+};
+
 const consoleFetchCommon = async (
   url: string,
   method: string = 'GET',
   options: RequestInit = {},
   timeout?: number,
-) => {
+  isEntireResponse?: boolean,
+): Promise<Response> => {
```

Comment:
Can't this function also return JSON or text?

Reply:
> Can't this function also return JSON or text?

Yes, text but for JSON I do not see a need for it. Because it could be any structure, and the `response` passed into `parseData` is of type `Response`.